core/src/main/java/org/hibernate/ogm/dialect/spi/AssociationTypeContext.java

```diff
@@ -20,13 +20,20 @@
 public interface AssociationTypeContext {
 
 	/**
-	 * Provide access to the options set for an entity or property.
+	 * Provide access to the options set for the association.
 	 *
-	 * @return a context object providing access to the options effectively applying for a given entity or property.
+	 * @return a context object providing access to the options effectively applying for the association.
 	 */
 	OptionsContext getOptionsContext();
 
 	/**
+	 * Provide access to the options set for the entity owner of the association.
+	 *
+	 * @return a context object providing access to the options effectively applying for the owner entity.
+	 */
+	OptionsContext getOwnerEntityOptionsContext();
```

Comment:
Could this info not be exposed via the existing `getOptionsContext()`. The idea originally was that this exposes the effectively applying options for a given element. My concern is that the two separate methods further complicate things for dialect implementations.